HIP/hip-514-missing-hts-precompile-functions.md

```diff
@@ -132,36 +132,36 @@ by the specific precompile function being called (see the functions below).
-| ``        | `wipeTokenAccount(address, address, [uint64])`    | `([uint64])`              |
-| ``        | `updateTokenInfo(address, HederaToken)`           | `(HederaToken)`           |
-| ``        | `updateTokenAutoRenewInfo(address, uint32)`       | `(address, uint32)`       |
-| ``        | `updateTokenKeys(address, uint, KeyValue)`        | `(TokenKey[])`            |
+| hash        | effective signature                               | return                    |
+| ------------|---------------------------------------------------|---------------------------|
+| `927da105`  | `allowance(address, address, address)`            | `(uint256)`               |
+| `e1f21c67`  | `approve(address, address, uint256)`              | `(uint256)`               |
+| `7336aaf0`  | `approveNFT(address, address, uint256)`           | `(uint256)`               |
+| `098f2366`  | `getApproved(address, uint256)`                   | `(uint256, address)`      |
+| `f49f40db`  | `isApprovedForAll(address, address, address)`     | `(uint256, bool)`         |
+| `367605ca`  | `setApprovalForAll(address, address, bool)`       | `(uint256)`               |
+| `46de0fb1`  | `isFrozen(address, address)`                      | `(bool)`                  |
+| `f2c31ff4`  | `isKyc(address, address)`                         | `(bool)`                  |
+| `f069f712`  | `deleteToken(address)`                            | `(bool)`                  |
+| `15c981cb`  | `getTokenAutoRenewInfo(address)`                  | `(address, uint32)`       |
+| `ae7611a0`  | `getTokenCustomFees(address)`                     | `(CustomFee[])`           |
+| `a7daa18d`  | `getTokenDefaultFreezeStatus(address)`            | `(bool)`                  |
+| `335e04c1`  | `getTokenDefaultKycStatus(address)`               | `(bool)`                  |
+| `3f28a19b`  | `getFungibleTokenInfo(address)`                   | `(FungibleTokenInfo)`     |
+| `1f69565f`  | `getTokenInfo(address)`                           | `(TokenInfo)`             |
+| `8946344a`  | `getTokenKey(address, uint)`                      | `(KeyValue)`              |
+| `2c20dcd1`  | `getNonFungibleTokenInfo(address, uint32)`        | `(NonFungibleTokenInfo)`  |
+| `5b8f8584`  | `freezeToken(address, address)`                   | `(bool)`                  |
+| `a9b86b17`  | `unFreezeToken(address, address)`                 | `(bool)`                  |
+| `8f8d7f99`  | `grantTokenKyc(address, address)`                 | `(bool)`                  |
+| `af99c633`  | `revokeTokenKyc(address, address)`                | `(bool)`                  |
+| `7c41ad2c`  | `pauseToken(address)`                             | `(bool)`                  |
+| `3f8cc5c2`  | `unPauseToken(address)`                           | `(bool)`                  |
+| `9790686d`  | `wipeTokenAccount(address, address, uint32)`      | `(uint32)`                |
+| `1da62b01`  | `wipeTokenAccount(address, address, [uint64])`    | `([uint64])`              |
```

Comment:
Isn't it better here to have only one function for wipeTokenAccount(address, address, uint32, uint64[])?
In this way we will be consistent with previously implemented precompiles (mint, burn).

Reply:
Prefer the separate methods as is cleaner. 
I think it's okay to use this as the improved established standard

Reply:
Fair point `wipeTokenAccount` for NFTs has different parameters so it's explicit.
Approve for fungible and non fungible had the same signature so there was a need to differentiate so we could differentiate an amount vs a serial number.

The return in a successful operation will be the same but like other update/delete operations the goal is to provide a validation of the change to the caller

Reply:
Imo the response code from the execution of the precompile is enough to provide a validation of the change to the caller.

Reply:
I'm convinced now and agree. I've dropped it from the solidity files and will update here also

Reply:
Updated